### DISCUSSION TITLE
Phalcon 4.0.5 Slow on certain hasMany relationship queries

### DISCUSSION THREAD

**Discussion by davidbiga** [phalcon/cphalcon#15924] [Q&A]
Hey friends,

Love everything about Phalcon.  So thank you to everyone who contributes.

I have an old Docker image running `PHP Version 7.0.30-1+ubuntu16.04.1+deb.sury.org+1` with `Phalcon version 3.4.0` that was upgraded to a new server with Phalcon version `4.0.5` running on `5.10.76-linuxkit` - from ` php:7.4-apache`.

I can't for the life of me figure out why I have some tables that do not respond from the `hasMany` relationships on certain models and looking to get anyones input, suggestions.

Here is what happens:

I have `Model A` -> hasMany `Model Bs`.

When I would `find()` `Model A's` and reference `modelA->modelBParts` in something like a foreach loop for instance, it'll just hang forever and sometimes max out memory (500mb).  

Now for other relationships like `Model C -> Model D`, no issues with the hasMany.  I've looked to confirm my MySql database is index in that area and looked at some of the PDO settings as well.  I've even tried direct psql.  

Thank you again for your time.

**Comment by BeMySlaveDarlin** [comment]
Check out please recommendations in discussion #15915
